### ISSUE TITLE
Streaming is broken

### ISSUE THREAD

**Issue by randomsnowflake** [MacPaw/OpenAI#14]
**Code**
```swift
import Foundation
import OpenAI

let openAI = OpenAI(apiToken: "sk-...")

let query = OpenAI.ChatQuery(model: .gpt3_5Turbo, messages: [.init(role: "user", content: "hi!")], stream: true)
let result = try await openAI.chats(query: query)
```

**Exception**
```
Swift/ErrorType.swift:200: Fatal error: Error raised at top level: Swift.DecodingError.dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid JSON.", underlyingError: Optional(Error Domain=NSCocoaErrorDomain Code=3840 "Invalid value around line 1, column 0." UserInfo={NSDebugDescription=Invalid value around line 1, column 0., NSJSONSerializationErrorIndex=0})))

2023-03-08 14:37:01.246548+0100 CLIChat[81308:5267103] Swift/ErrorType.swift:200: Fatal error: Error raised at top level: Swift.DecodingError.dataCorrupted(Swift.DecodingError.Context(codingPath: [], debugDescription: "The given data was not valid JSON.", underlyingError: Optional(Error Domain=NSCocoaErrorDomain Code=3840 "Invalid value around line 1, column 0." UserInfo={NSDebugDescription=Invalid value around line 1, column 0., NSJSONSerializationErrorIndex=0})))
```

It works with `stream: false`

**Comment by Krivoblotsky** [comment]
Hey, @randomsnowflake.
Thanks for bringing this to our attention. 
Looks like streaming is unsupported for now and we need to implement server-sent events for this.

**Comment by longseespace** [comment]
I hope this will be implemented soon. Meanwhile, you can use this alternative (only for streaming - I'm using both clients in my app)

https://github.com/nate-parrott/openai-streaming-completions-swift

**Comment by randomsnowflake** [comment]
I already use it. I think it's also the only Swift OpenAI API with streaming support. 

However, have a look at my issue report here: [Missing Model Parameters and No Error Message](https://github.com/nate-parrott/openai-streaming-completions-swift/issues/3) ... it's not as mature as I would like. Looking forward to MacPaw's implementation of streaming.

**Comment by Krivoblotsky** [comment]
I know that @rshimoda is on his way to adding streaming support here, so I believe it must be implemented shorty 👌

**Comment by DJBen** [comment]
@rshimoda Hi Sergi do you have a timeline for the implementation? Let me know if you need any help or have any WIP branch.

**Comment by Krivoblotsky** [comment]
@rshimoda 🙏

**Comment by DJBen** [comment]
I am exploring a solution right now - it looks like a bigger work than I expect, and it is more akin to a separate project than 'fixing' the streaming.

We might need to rely on other `EventSource` libraries like https://github.com/launchdarkly/swift-eventsource. Let me know if you think is okay to add dependency to another library.

**Comment by Krivoblotsky** [comment]
Hey, @DJBen and everyone. 
I have some implementation in mind that doesn't require any 3rd party libraries. I have implemented PoC that supports streaming for Chats API with minimal changes required. 
Could you take a look at https://github.com/MacPaw/OpenAI/pull/57?

It has several weak points, such as optional delta and message in ChatResult, but overal looks good to me. 
Public interfaces is untouched, but the completion coulbe called multiple times with the delta:

```swift
openAI.chats(query: .init(model: .gpt3_5Turbo, messages: [.init(role: .user, content: "Who is Taras Shevchenko?")], stream: true)) { result in
    switch result {
    case .success(let res):
        //called multiple times
        if let messageDelta = res.choices.first?.delta?.content {
            print(messageDelta)
        }
    case .failure(let error):
        print(error)
    }
}
```
Any feedback is very appreciated 🙏

**Comment by Krivoblotsky** [comment]
Almost there: #57 🤞

**Comment by Krivoblotsky** [comment]
Fixed by #57
